docs/README-encryption.md

```diff
@@ -0,0 +1,57 @@
+# Data-at-rest encryption
+
+Encryption is the process of encoding information in a format that cannot be read or understood by an eavesdropper, and it might be a required feature for use cases where sensitive data needs to remain confidential and cannot be accessed by unauthorized parties.
+
+Encryption is currently supported on the following SoMs:
+
+- Apalis iMX6
+- Apalis iMX8
+- Colibri iMX6DL
+- Colibri iMX6ULL (1GB eMMC variant only)
+- Colibri iMX7D (1GB eMMC variant only)
+- Colibri iMX8X
+- Verdin iMX8MM
+- Verdin iMX8MP
+
+## Data-at-rest encryption implementation
+
+Implementing data-at-rest encryption requires a mechanism to securely store the encryption key in the device, as well as an infrastructure to encrypt and decrypt data on disk at runtime.
+
+For this, the following features are implemented by this layer:
+
+- **Encryption key management**: the Linux kernel Trusted Keys subsystem is used to securely store and manage the private key used to encrypt and decrypt data on the disk.
+- **Block device encryption**: the Linux kernel `dm-crypt` subsystem is used to provide transparent disk encryption to users.
+
+## Encryption key management
+
+One of the main challenges to implementing encryption is how to securely store the encryption key. If the key is compromised, the encrypted data can be easily decrypted.
+
+To solve this, a Linux kernel feature called [Trusted Keys](https://docs.kernel.org/security/keys/trusted-encrypted.html) is used.
+
+Trusted Keys make it possible to create and manage variable-length symmetric keys in kernel space, and user space only sees, stores, and loads encrypted blobs.
+
+Trusted Keys require the availability of a Trust Source for greater security, and CAAM is leveraged on iMX-based SoMs. TPM (Trusted Platform Module) and TEE (Trusted Execution Environment) are also two other possible trust sources for Trusted Keys, but currently not supported by this layer.
+
+## Block device encryption
+
+The Linux kernel [dm-crypt](https://docs.kernel.org/admin-guide/device-mapper/dm-crypt.html) subsystem is used to provide transparent disk encryption to users.
+
+`dm-crypt` is part of the device-mapper framework and works by creating a virtual block device that encrypts data as it is written to the disk and decrypts it as it is read.
+
+## Enabling and configuring encryption
```

Comment:
Should we also mention in this section that `TDX_ENC_STORAGE_LOCATION` must be explicitly set to some valid value?

Reply:
Makes sense! PR updated.